plugin/rewrite/README.md

```diff
@@ -151,6 +151,62 @@ ftp-us-west-1.coredns.rocks. 0    IN A    10.20.20.20
 ftp-us-west-1.coredns.rocks. 0    IN A    10.30.30.30
 ```
 
+Note that names in SRV Record Targets and records in the `ADDITIONAL SECTION` 
+will also be rewritten following the `answer name` rule to keep the response 
```

Comment:
> and records in the `ADDITIONAL SECTION` will also be rewritten

but not always, only for SRV requests (for now)...

"For SRV responses, SRV targets and records in the `ADDITIONAL SECTION`  will also be rewritten ..."

I don't think the following examples are necessary in context with the prior examples, it should be clear what this means.

Reply:
hmm - i just re-read the code - and it looks like it may _always_ be rewriting the additional section?  If so, that will break some responses where targets no longer match the additional section entries.

Reply:
Agreed the wording here is difficult.

But I dont think we can limit the description to just "For SRV responses..." consider the following answer/additional sections:

```
;; ANSWER SECTION:
_test._tcp.test-service.cluster.local. 15 IN SRV 0 50 1234 10-1-1-2.test-service.cluster.local.

;; ADDITIONAL SECTION:
10-1-1-2.test-service.cluster.local. 15 IN A 10.1.1.2
```

The code in this PR will always rewrite the names on the "left hand side" of both the answer and additional sections regardless of if the additional section relates to a SRV answer or something else.
However, you are correct that for names on the "right hand side" that they will only be rewritten for SRV (for now).

I'll try and come up with something better (suggestions are welcome).

Reply:
> The code in this PR will always rewrite the names on the "left hand side" of both the answer and additional sections regardless of if the additional section relates to a SRV answer or something else.

OK.  IIUC, this will break some responses, causing named targets in non-SRV records not to match the entries in the additional section.

Reply:
Resolving this as we have changed direction based on the discussion below and in the issue thread.